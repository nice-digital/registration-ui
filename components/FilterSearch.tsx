import React from "react";
import { Panel } from "@nice-digital/nds-panel";
import { Input } from "@nice-digital/nds-forms";

type FilterSearchProps = {
	onInputChange: (searchQuery: string) => void;
	label: string;
};

export default function FilterSearch({onInputChange, label}: FilterSearchProps) {
	let typingTimer = 0;

	const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
		const val = e.target.value.trim();

		clearTimeout(typingTimer);

		typingTimer = window.setTimeout(() => {
			onInputChange(val);
		}, 1000);
	};

	return (
		<Panel>
			<Input
				type="search"
				label={label}
				unique="filterSearch"
				name="filter-search"
				onChange={handleInputChange}
				autoComplete="off"
				data-qa-sel="filter-search-input"
			/>
		</Panel>
	);
};
