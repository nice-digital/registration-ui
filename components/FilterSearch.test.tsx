import React from "react";
import { render, screen } from "@testing-library/react";
import FilterSearch from "./FilterSearch";

const props = {
	onInputChange: jest.fn(),
	label: "Filter by topic name"
};

describe("FilterSearch", () => {
	it("renders a heading", () => {

		render(<FilterSearch {...props} />);

		const heading = screen.getByText("Something that isn't the correct label", {
			selector: ".input__label",
		})

		expect(heading).toBeInTheDocument();
	});

	it("should match snapshot for main content", () => {
		render(<FilterSearch {...props} />);

		expect(document.body).toMatchSnapshot();
	});
})