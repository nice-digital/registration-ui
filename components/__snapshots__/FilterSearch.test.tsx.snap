// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSearch should match snapshot for main content 1`] = `
<body>
  <div>
    <div
      class="panel"
    >
      <div
        class="input"
      >
        <label
          class="input__label"
          for="filter-search"
        >
          Filter by topic name
        </label>
        <input
          autocomplete="off"
          class="input__input"
          data-qa-sel="filter-search-input"
          id="filter-search"
          name="filter-search"
          type="search"
          unique="filterSearch"
          value=""
        />
      </div>
    </div>
  </div>
</body>
`;
